cmd/osm-bootstrap/crds/policy_upstream_traffic_setting.yaml

```diff
@@ -225,6 +461,138 @@ spec:
                                       description: Value defines the HTTP header value.
                                       type: string
                                       minLength: 1
+                          global:
+                            description: Global rate limiting policy applied per route.
+                            type: object
+                            required:
+                            - descriptors
+                            properties:
+                              descriptors:
+                                description: List of rate limit descriptors to use in the rate limit service request.
+                                type: array
+                                minItems: 1
+                                items:
+                                  description: Rate limit descriptor to use in the rate limit service request.
+                                  type: object
+                                  required:
+                                  - entries
+                                  properties:
+                                    entries:
```

Comment:
For each "entries" you can only set one of the following: genericKey, remoteAddress, requestHeader, headerValueMatch. Would it make sense for "entries" to be "entry"?

Reply:
No, multiple entries are allowed per descriptor. Entries and each entry needs to exactly map to the Go type schema: https://github.com/openservicemesh/osm/blob/main/pkg/apis/policy/v1alpha1/upstreamtrafficsetting.go#L320.

What's your concern here?

Reply:
I understand that multiple entries can be specified per descriptor. My concern is that the plural "entries" is misleading given only one of the following fields can be set per entry: enericKey, remoteAddress, requestHeader, headerValueMatch.

```
        descriptors:
          # generates descriptor entry [ remote_address=<client-ip from x-forwarded-for header> ]
          - entries:
            - remoteAddress: {}
          - entries:
            # generates descriptor entry [ foo=bar ]
            - genericKey:
                key: foo # optional, default: generic_key
                value: bar
          - entries:
            # generates descriptor entries [ header_match=get, service=<x-service header value> ]
            - headerValueMatch:
              key: header_match # optional, default: header_match
              value: get
              expectMatch: true # optional, default: true
              headers:
                - name: ":method"
                  prefix: /get
```

Reply:
I think there may be a misunderstanding here. The plural `entries` makes sense because there can be multiple descriptor entries as follows:
```
          - entries:
            - genericKey:
                key: foo
                value: bar
            - requestHeader:
                name: x-service
                key: service
            - headerValueMatch:
                key: header_match
                value: get
                expectMatch: true
                headers:
                  - name: ":method"
                    prefix: /get
```

Think of the descriptor entry being polymorphic while being one of `genericKey`, `requestHeader`, `headerValueMatch`, `remoteAddress`.

Reply:
Got it. I misunderstood what the `oneOf` applied to. Thanks!